File: app/src/main/java/com/example/volleyrvpractice/Recipe/JsonData2Recipe.java
package com.example.volleyrvpractice.Recipe;

import com.example.volleyrvpractice.FavouriteRecipeModel.FavouriteRecipe;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class JsonData2Recipe {
    public static String subImageUrl1 = "https://spoonacular.com/recipeImages/";
    public static String subImageUrl2 = "-556x370.jpg";

    public static List<RecipeModel> jsonObject2RandomRecipe(JSONObject jsonObject, List<FavouriteRecipe> fRList) throws JSONException {
        List<RecipeModel> list = new ArrayList<>();
        JSONArray resultsArray = jsonObject.getJSONArray("recipes");

        for(int i =0; i<resultsArray.length();i++){
            JSONObject recipeData = (JSONObject) resultsArray.get(i);
            String recipe_id = recipeData.getString("id");
            String recipe_title = recipeData.getString("title");
            String recipe_imageUrl = subImageUrl1+recipeData.getString("id")+subImageUrl2;

            boolean fRIndicator = false;
            for(FavouriteRecipe fR: fRList){
                if(fR.getId().equals(recipe_id)){
                    fRIndicator = true;
                }
            }

            list.add(new RecipeModel(recipe_id, recipe_title, recipe_imageUrl, 0, fRIndicator, null));
        }

        return list;
    }

    public static List<RecipeModel> jsonObject2SearchRecipe(JSONObject jsonObject, List<FavouriteRecipe> fRList) throws JSONException {
        List<RecipeModel> list = new ArrayList<>();
        JSONArray resultsArray = jsonObject.getJSONArray("results");

        for(int i =0; i<resultsArray.length();i++){
            JSONObject recipeData = (JSONObject) resultsArray.get(i);
            String recipe_id = recipeData.getString("id");
            String recipe_title = recipeData.getString("title");
            String recipe_imageUrl = subImageUrl1+recipeData.getString("id")+subImageUrl2;

            boolean fRIndicator = false;
            for(FavouriteRecipe fR: fRList){
                if(fR.getId().equals(recipe_id)){
                    fRIndicator = true;
                }
            }

            list.add(new RecipeModel(recipe_id, recipe_title, recipe_imageUrl, 0, fRIndicator, null));
        }

        return list;
    }
}

File: app/src/main/java/com/jasonstudio/cookbook2/Recipe/JsonData2Recipe.kt
package com.jasonstudio.cookbook2.Recipe

import com.jasonstudio.cookbook2.FavouriteRecipeModel.FavouriteRecipe
import org.json.JSONException
import org.json.JSONObject

object JsonData2Recipe {
    @JvmField
    var subImageUrl1 = "https://spoonacular.com/recipeImages/"
    @JvmField
    var subImageUrl2 = "-556x370.jpg"
    @JvmStatic
    @Throws(JSONException::class)
    fun jsonObject2RandomRecipe(
        jsonObject: JSONObject,
        fRList: List<FavouriteRecipe>
    ): List<RecipeModel> {
        val list: MutableList<RecipeModel> = ArrayList()
        val resultsArray = jsonObject.getJSONArray("recipes")
        for (i in 0 until resultsArray.length()) {
            val recipeData = resultsArray[i] as JSONObject
            val recipe_id = recipeData.getString("id")
            val recipe_title = recipeData.getString("title")
            val recipe_imageUrl = subImageUrl1 + recipeData.getString("id") + subImageUrl2
            var fRIndicator = false
            for (fR in fRList) {
                if (fR.id == recipe_id) {
                    fRIndicator = true
                }
            }
            list.add(RecipeModel(recipe_id, recipe_title, recipe_imageUrl, 0, fRIndicator, null))
        }
        return list
    }

    @JvmStatic
    @Throws(JSONException::class)
    fun jsonObject2SearchRecipe(
        jsonObject: JSONObject,
        fRList: List<FavouriteRecipe>
    ): List<RecipeModel> {
        val list: MutableList<RecipeModel> = ArrayList()
        val resultsArray = jsonObject.getJSONArray("results")
        for (i in 0 until resultsArray.length()) {
            val recipeData = resultsArray[i] as JSONObject
            val recipe_id = recipeData.getString("id")
            val recipe_title = recipeData.getString("title")
            val recipe_imageUrl = subImageUrl1 + recipeData.getString("id") + subImageUrl2
            var fRIndicator = false
            for (fR in fRList) {
                if (fR.id == recipe_id) {
                    fRIndicator = true
                }
            }
            list.add(RecipeModel(recipe_id, recipe_title, recipe_imageUrl, 0, fRIndicator, null))
        }
        return list
    }
}